### PR TITLE
:wrench: Add fields prescription

### PR DESCRIPTION
**Pull request by issa-diallo**

### CODE REVIEW

**Comment by issa-diallo** [comment]
The file and swicth fields are the object of another PR.

File: src/components/PatientForm.tsx [hunk 1]
```diff
@@ -64,78 +78,166 @@ const PatientForm: FunctionComponent<PatientFormProps> = ({ patient }) => {
...
-          />
-        </Form.Group>
-      </Row>
-
-      <Button variant='primary' type='submit'>
-        Valider
-      </Button>
-      <Button onClick={onDelete} className='ms-4' variant='danger'>Supprimer</Button>{' '}
-    </Form>
+    <>
```
Comment (AndreMiras) [inline src/components/PatientForm.tsx:81]: I don't think we need a fragment here

File: src/components/PatientForm.tsx [hunk 2]
```diff
@@ -25,10 +25,24 @@ const PatientForm: FunctionComponent<PatientFormProps> = ({ patient }) => {
...
-    const newField = { [fieldName]: fieldValue }
-    setPatientState({ ...patientState, ...newField })
+    const keys = e.target.name.split('.')
+    const fieldName = keys[0]
+    if (fieldName in patientState) {
+      const field = patientState[fieldName as keyof typeof patientState]
+      if (field instanceof Object) {
+        const nextValue = [...field]
+        // @ts-expect-error
+        nextValue[keys[1]][keys[2]] = fieldValue
```
Comment (AndreMiras) [inline src/components/PatientForm.tsx:36]: Looks very fishy.
Maybe for the prescription you want to create a dedicated state if that makes it easier to manage

Comment (issa-diallo) [reply src/components/PatientForm.tsx:36]: yes I wanted a dedicated state but in doubt of your answer I made a single state

File: src/components/PatientForm.tsx (hunk 1 above)
Comment (issa-diallo) [reply src/components/PatientForm.tsx]: fragment deleted

**Review by AndreMiras — APPROVED** [review]
There're a few flags, I think things can be simplified.
Also cut into more components e.g. for prescription.
Let's not block it, but follow up with clean up after

File: src/components/PatientForm.tsx (hunk 2 above)
Comment (AndreMiras) [reply src/components/PatientForm.tsx:36]: Fixed in https://github.com/issa-diallo/mynotif_frontend/pull/114

**Pull request merged**